source/use-cases/sensitive-data-encryption.txt

```diff
@@ -57,12 +57,122 @@ cubilia Curae; Nunc non interdum purus, ultricies laoreet tortor.
+            }
+        }
+
+Create JSON objects for each field and append them to the `properties` map.
+
+.. code::
+
+    // Deterministic encryption supports querying and is equally as secure as random encryption for high cardinality (very unique) fields such as SSN. A privileged user in the MedcoMD Medical Care Management System might want to query on SSN to verify a patient's identity. One thing to note is that you do not have to include keyId if you specified it in the top-level encryptMetadata field. Though you could include it if you want different users with specific keys to be able to access a specific subset of the encrypted data.
+
+    "ssn": {
+        "encrypt": {
+            "bsonType": "int",
+            "keyId": <keyID>,
+            "algorithm": "AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic"
+        }
+    }
+
+
+.. code:: javascript
+
+    // Random encryption does not support filtering and is best used on fields with low cardinality (very few unique fields) to prevent against frequency analysis attacks. Since there are only 8 blood types, this field is randomly encrypted. To decrypt this data, a privileged user would have to query on another field via a secure collection, and this data would be returned unencrypted.
+
+    "bloodType": {
+        "encrypt": {
+            "bsonType": "string",
+            "keyId": <keyId>,
+            "algorithm": "AEAD_AES_256_CBC_HMAC_SHA_512-Random"
+        }
+    }
+
+
+.. code:: javascript
+
+    // Arrays can only be auto-encrypted via random encryption. 
+
+    "medicalRecords": {
+        "encrypt": {
+            "bsonType": "array",
+            "keyId": <keyID>,
+            "algorithm": "AEAD_AES_256_CBC_HMAC_SHA_512-Random"
+        }
+    }
+
+.. code:: javascript
+
+    // insurance.policyNumber number is deterministically encrypted because the policy number is a very unique field. Note, that the JSON for policyNumber follows the same structure the JSON for SSN, however, policyNumber is nested inside insurance.properties because it is a nested field.
+
+    "insurance": {
+        "bsonType": "object",
+        "properties": {
+            "policyNumber": {
+                "encrypt": {
+                    "bsonType": "int",
+                    "keyId": <keyID>,
+                    "algorithm": "AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic"
+                }
+            }
+        }
+    }    
+
+
+
+- View the full JSON Schema for the Medco Medical Management System `here <https://raw.githubusercontent.com/mongodb/docs-assets/DOCSP-json-schema-helper-and-json/MedcoMDSchema.json>`_
+- View the `helper code <https://raw.githubusercontent.com/mongodb/docs-assets/DOCSP-json-schema-helper-and-json/FieldLevelEncryptionJSONSchema.java>`_ to generate the full JSON Schema for this MedcoMD guide in Java. 
+
+.. note::
+
+    this helper code is not the full code for this guide; it only contains the code to generate a JSON Schema for this section
```

Comment:
In your previous `.. note::` directive, you started the sentence with a capitalized letter and ended with a period -- it would be great if we could standardize. I prefer the caps + period at the end, but can adjust all the instances of my own for this guide if there are any reasons to do it differently.

Reply:
Also, do you think we need this note? It's clear to me from the mention of "View the helper code to generate the full JSON Schema...".

Reply:
Will comment on the updated commit.

Reply:
* To address your first concern: valid point, I've changed this note to be capitalized and end with a period. 
* To address your second concern: I'm neutral to the note, but it was your suggestion originally "I meant a formatted note, e.g.:

.. note::
   This helper code is not the full..."

 perhaps @nlarew could weigh his opinion here.

Reply:
Yes, my current suggestion is to omit the note completely.

Reply:
Fixed, it's omitted now